eth/db/diff.py

```diff
@@ -124,12 +129,42 @@ def __getitem__(self, key: bytes) -> bytes:
 
     def __iter__(self) -> None:
         raise NotImplementedError(
-            "Cannot iterate through changes, use apply_to(db) to update a database"
+            "Cannot iterate through changes, use apply_to(db) to update a database. "
+            "Also, new_keys(), deleted_keys(), and new_items() might be of interest."
         )
 
     def __len__(self) -> int:
         return len(self._changes)
 
+    @to_tuple
+    def deleted_keys(self) -> Iterable[bytes]:
+        """
+        List all the keys that have been deleted.
+        """
+        for key, value in self._changes.items():
+            if value is DELETED:
+                yield key
+
+    @to_tuple
+    def new_keys(self) -> Iterable[bytes]:
```

Comment:
Just noting that the name is slightly confusing because at first sight it makes it look as if this (and similarly `new_items`) would only contain *additions*  when in reality the `DiffTracker` simply does not differentiate between *updated*  or *inserted* keys. I think it can not differentiate between these two cases but maybe the API name can be clearer to reflect that?

I'm not the best with naming suggestions but I'll try nonetheless:

- `set_keys` / `set_items`
- `touched_keys` / `touched_items`
- `new_and_updated_keys` / `new_and_updated_items`

Reply:
Agreed. Do you think that `updated_keys()` has the opposite problem?

If you don't like `updated_keys()` or `changed_keys()`, how about `pending_keys()`, `dirty_keys()` or `new_value_keys()`?

`set` has the problem that it sounds like a verb. `touched` could also apply to deleted keys, and `new_and_updated` is long. Just hoping we can find a magic name :)

Reply:
>Do you think that `updated_keys()` has the opposite problem?

Yes, kind of. I think I like `pending` and `dirty` better :thumbsup:

Reply:
`pending` it is :+1: